src/components/basic-element/types/textedit.js

```diff
@@ -159,7 +158,7 @@ module.exports = {
       if(!this.state.editing) {
         this.onTextUpdate({
           target: {
-            value: this.props.innerHTML.trim()
```

Comment:
Most of it was cleaning up, though the `white-space` stuff appears to have been unnecessary and the cause of the issue.